language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/util/ServiceTemplateGenerator.java

```diff
@@ -110,50 +126,111 @@ private ServiceTemplateGenerator(LanguageServerContext context) {
 
                 // Start progress
                 WorkDoneProgressBegin beginNotification = new WorkDoneProgressBegin();
-                beginNotification.setTitle(TITLE_INITIALIZE);
+                beginNotification.setTitle(TITLE_SERVICE_TEMPLATE_GENERATOR);
                 beginNotification.setCancellable(false);
                 beginNotification.setMessage("Initializing...");
                 languageClient.notifyProgress(new ProgressParams(Either.forLeft(taskId),
                         Either.forLeft(beginNotification)));
             }
-
-        }).thenRunAsync(() -> initialize(context)).thenRunAsync(() -> {
-            clientLogger
-                    .logTrace("Finished loading listener symbols from the distribution");
+        }).thenRunAsync(() -> {
+            try {
+                this.loadListenersFromDistribution(lsContext);
+            } catch (Throwable e) {
+                //ignore
+                clientLogger.logTrace("Failed loading listener symbols from the  BallerinaUserHome due to "
```

Comment:
Shouldn't this part go to `exceptionally()` part and inside that, we should show the notification as `failed` as well? Otherwise always the message in progress will be a success message.

Reply:
Done